subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,112 @@
+project('speexdsp',
+        'c',
+        version : '1.2.0',
+        license : 'BSD-3-Clause',
+        default_options : [
+          'c_std=c99',
+          'default_library=both',
+          'warning_level=0'])
+
+# library versioning from speexdsp-1.2.0's configure.ac
+# SPEEXDSP_LT_CURRENT=6, SPEEXDSP_LT_REVISION=1, SPEEXDSP_LT_AGE=5
+lib_current = 6
+lib_revision = 1
+lib_age = 5
+lib_soversion = lib_current - lib_age
+lib_version = '@0@.@1@.@2@'.format(lib_soversion, lib_age, lib_revision)
+
+c_args = []
+cc = meson.get_compiler('c')
+host_cpu_family = host_machine.cpu_family()
+
+# Round up configuration data
+add_project_arguments('-DHAVE_CONFIG_H',
+                      language: 'c')
+cdata = configuration_data()
+
+# Set SIMD config data and flags
+simd_type = 'None'
+if host_cpu_family.startswith('x86')
+  cdata.set('USE_SSE', true)
+  cdata.set('USE_SSE2', true)
+  c_args += ['-mfpmath=sse',
```

Comment:
mfpmath is not set upstream, just -msse -msse2... dunno if this matters.

mfpmath=sse is also the default for x86_64 anyway. :)

Reply:
Ok, let's tighten this up!

SSE and SSE2 are always available on x64-64 CPUs, so we don't need flags. So I'm now handing x86-64 on its own and merely toggle the `cdata` items on to let the code use explicit SSE intrinsic code paths:

For 32bit x86, we do need `-mfpmath=sse`, because it's the only way to both enable and use SSE extensions. Ref: https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html.  

If cmake is missing this, then I suppose they're assuming the user is externally adding this flag.. otherwise no SSE for them :sweat_smile:

Reply:
- `-msse -msse2` "allows" the build to use intrinsics.
- `-mfpmath=sse` I think just enables automatic conversion of codegen to prefer intrinsics. Either way if you manually code the intrinsics yourself, they will be used.

If the library requires SSE anyway, I guess there's no reason not to set -mfpmath=sse though, it's pure benefit. So that makes sense.

Thanks for the clarification and code comments.